app/models/base.py

```diff
@@ -40,3 +45,54 @@ class Config:
         match their type annotations"""
 
         validate_assignment = True
+
+
+class BaseResponse(TableModelMixin):
+    pass
+
+
+class BaseRequest(BaseModel):
+    class Meta:
+        related_fields = {}
+        model = None
+
+    def get_model(self):
+        if not self.Meta.model:
+            raise NotImplementedError(
+                "Either set the Meta.model property or override the get_model() method."
+            )
+        return self.Meta.model
+
+    def get_related_fields(self):
+        return self.Meta.related_fields
+
+    def translate(self):
+        """Convert a dump of request to a dict that can easily be used to create an instance of a model"""
+        data = self.model_dump()
+        related_fields_names = self.get_related_fields()
+        related_fields_data = {}
+        fields = {}
+        for field_name, field_value in data.items():
+            if field_name in related_fields_names:
+                related_fields_data[field_name] = field_value
+            else:
+                fields[field_name] = field_value
+        return {**fields, **self._translate_related_fields(related_fields_data)}
+
+    def _translate_related_fields(self, fields):
```

Comment:
How deep does this go? If we have a model with a sub model, with a sub model will it be able to handle this?

For example, I think we could have `Person.Eligibility.Income`.

Should we support this?

Reply:
We can wait until there is need, however, I think there should be a comment added to the translate method that this currently only supports single depth relationships.

Reply:
Updated the doc string for `app.models.base.BaseRequest.translate` to reflect this current limitation